### PR TITLE
Ford documentation

### PR DESCRIPTION
**Pull request by kmorris-si**
# Functionality added in this branch
- [x] Add build instructions in readme
- [x]  Add ford configuration file
- [x]  Add CI for Ford documentation. At this time it generates a zip file with all html files

### CODE REVIEW

File: doc/miniFAVOR-FORD-file.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+coloured_edges: true
+sort: permission-alpha
+extra_mods: iso_fortran_env:https://gcc.gnu.org/onlinedocs/gfortran/ISO_005fFORTRAN_005fENV.html
+            iso_c_binding:https://gcc.gnu.org/onlinedocs/gfortran/ISO_005fC_005fBINDING.html#ISO_005fC_005fBINDING
+lower: true
+print_creation_date: true
+creation_date: %Y-%m-%d %H:%M %z
+project_github: https://github.com/everythingfunctional/miniFAVOR
+project_download: hhttps://github.com/everythingfunctional/miniFAVOR
+github: https://github.com/everythingfunctional/miniFAVOR
```
Comment (everythingfunctional) [inline doc/miniFAVOR-FORD-file.md:27]: The above 3 links should be changed to sourceryinstitute now

File: doc/miniFAVOR-FORD-file.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Getting Help
+------------
+
+If you encounter a problem, have a suggestion, or want to ask a question,
+we encourage you to post an issue in [this projects Github repository] by
+[opening a new issue]. 
+
+[this projects Github repository]: https://github.com/everythingfunctional/miniFAVOR
+[opening a new issue]: https://github.com/everythingfunctional/miniFAVOR/issues/new
```
Comment (everythingfunctional) [inline doc/miniFAVOR-FORD-file.md:75]: These links should be changed to sourceryinstitute

File: doc/miniFAVOR-FORD-file.md (hunk 1 above)
Comment (kmorris-si) [reply doc/miniFAVOR-FORD-file.md]: The requested changes have been completed

File: doc/miniFAVOR-FORD-file.md (hunk 2 above)
Comment (kmorris-si) [reply doc/miniFAVOR-FORD-file.md]: The requested changes have been completed

**Review by everythingfunctional — COMMENTED** [review]
As discussed, recommend changing CI to publish to pages instead of creating a tarball

**Review by everythingfunctional — APPROVED** [review]
This looks good now.

**Pull request merged**